### PR TITLE
Signature display

### PR DESCRIPTION
**Pull request by sreynen**
For #18, this adds signature list display to petition pages.

### CODE REVIEW

File: src/components/signature-list-item.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const SignatureListItem = ({ signature, number }) => {
+  const { user } = signature
+  const { city, state } = user
+  const location = (city === '') ? state : `${city}, ${state}`
+  const fromLocation = (location === '') ? '' : ` from ${location}`
+  const date = new Date(signature.created_date)
+  return (
+    <li className='signer'>
+      <span className='signer-number'>{number}</span>
+      <b>{signature.user.name}</b> {fromLocation} signed this petition on {formatDate(date)}.
```
Comment (schuyler1d) [inline src/components/signature-list-item.js]: I don't see comments included in here.  Is that a todo?

Also, much more randomly, it looks like it's entirely driven by the server side, but just fyi, there's a petition option `hide_last_names` which means that name will only be first names.  Might be worth noting or passing the variable down into the component just for client-side consciousness (even if it doesn't affect the result).  Esp. if we ever try to separate the names.  probably disregard this, but I remembered something like this and took a while to hunt down, just to confirm that it IS ok to be server side.

Comment (sreynen) [reply src/components/signature-list-item.js]: Currently the client doesn't to know whether user.name is full name or not, but good flag if that changes.

File: src/components/signature-list.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    this.setState((prevState) => ({
+      page: prevState.page - 1
+    }))
+  }
+
+  render() {
+    const { loadSignatures, petitionSlug, signatures, signatureCount } = this.props
+    const { page } = this.state
+    if (typeof signatures === 'undefined' || typeof signatures[page] === 'undefined') {
+      loadSignatures(petitionSlug, page)
```
[the remark this answers is not in the document]
Comment (sreynen) [reply src/components/signature-list.js]: Added `loadSignaturesIfNeeded()` and called that from `componentDidMount()`, `previousPage()`, and `nextPage()`.

[the remark this answers is not in the document]
Comment (sreynen) [reply src/components/signature-list.js]: For the loading new signatures use case, I think we just call `loadSignatures()` for that, and that will update the store, which will update the `signatures` prop, which will cause it to re-render.

**Pull request merged**